pkg/planner/cardinality/selectivity.go

```diff
@@ -136,20 +136,23 @@ func Selectivity(
 				return 0, nil, errors.Trace(err)
 			}
 			nodes = append(nodes, &StatsNode{Tp: ColType, ID: id, mask: maskCovered, Ranges: ranges, numCols: 1})
+			var cnt float64
+			var cntEst statistics.RowEstimate
```

Comment:
```suggestion
			var cntEst *statistics.RowEstimate
```

I suggest creating a pointer here to avoid allocating memory once here and then again inside GetRowCountByColumnRanges.

Reply:
Discussed offline